src/record.rs

```diff
@@ -80,6 +80,52 @@ impl Meta {
     }
 }
 
+pub(crate) struct PartiallySerializedRecord {
+    buf: Vec<u8>,
+    offset_pos: usize,
+    header_checksum_pos: usize,
```

Comment:
`offset_pos` and `header_checksum_pos` are constants, so no need to store them as a part of `PartiallySerializedRecord`